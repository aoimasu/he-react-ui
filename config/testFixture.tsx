import * as React from 'react';
import createTestContext from 'react-cosmos-test/enzyme';

type Fixture = {
  component: React.ComponentType<any>;
  name?: string;
};

function testOneFixture(fixture: Fixture, variant: string = 'a') {
  const { mount, getWrapper } = createTestContext({ fixture });
  const { component } = fixture;
  const { displayName, name } = component as any;
  mount();
  test(`<${displayName || name} /> rendered correctly with ${fixture.name ||
    variant} fixture`, () => expect(getWrapper()).toMatchSnapshot());
}

// TODO: use real type guard functions to discriminate one fixture vs an array of them

export default function testAllFixtures(fixtures: any, variant?: string) {
  if (!('forEach' in fixtures)) {
    testOneFixture(fixtures as Fixture, variant);
  } else if (typeof fixtures.forEach === 'function') {
    (fixtures as Fixture[]).forEach(it => testOneFixture(it));
  }
}
