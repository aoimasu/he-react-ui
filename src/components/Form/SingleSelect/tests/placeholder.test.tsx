import fixture from '../fixtures/placeholder.fixture';
import testFixture from '../../../../../config/testFixture';

testFixture(fixture, 'placeholder');
