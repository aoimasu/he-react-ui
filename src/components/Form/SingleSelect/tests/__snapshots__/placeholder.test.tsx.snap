// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SingleSelect /> rendered correctly with placeholder fixture 1`] = `
<SingleSelect
  disabled={false}
  error=""
  id="demo"
  inline={false}
  name="demo"
  onBeforeOpen={[Function]}
  onChange={[Function]}
  onClose={[Function]}
  options={
    Array [
      Object {
        "label": "Option 1",
        "value": "1",
      },
      Object {
        "label": "Option 2",
        "value": "2",
      },
      Object {
        "label": "Option 3",
        "value": "3",
      },
    ]
  }
  placeholder="Select from the list"
  value={null}
>
  <div
    className="outer"
  >
    <div
      className="relative"
    >
      <div
        className="container noLabel"
      >
        <Select
          arrowRenderer={[Function]}
          autosize={true}
          backspaceRemoves={true}
          backspaceToRemoveMessage="Press backspace to remove {label}"
          className="select"
          clearAllText="Clear all"
          clearRenderer={[Function]}
          clearValueText="Clear value"
          clearable={true}
          closeOnSelect={true}
          deleteRemoves={true}
          delimiter=","
          disabled={false}
          escapeClearsValue={true}
          filterOptions={[Function]}
          id="demo"
          ignoreAccents={true}
          ignoreCase={true}
          inputProps={Object {}}
          isLoading={false}
          joinValues={true}
          labelKey="label"
          matchPos="any"
          matchProp="any"
          menuBuffer={0}
          menuRenderer={[Function]}
          multi={false}
          name="demo"
          noResultsText="No results found"
          onBlurResetsInput={true}
          onChange={[Function]}
          onClose={[Function]}
          onCloseResetsInput={true}
          onOpen={[Function]}
          onSelectResetsInput={true}
          openOnClick={true}
          optionComponent={[Function]}
          options={
            Array [
              Object {
                "label": "Option 1",
                "value": "1",
              },
              Object {
                "label": "Option 2",
                "value": "2",
              },
              Object {
                "label": "Option 3",
                "value": "3",
              },
            ]
          }
          pageSize={5}
          placeholder="Select from the list"
          removeSelected={true}
          required={false}
          rtl={false}
          scrollMenuIntoView={true}
          searchable={true}
          simpleValue={false}
          tabSelectsValue={true}
          trimFilter={true}
          value={null}
          valueComponent={[Function]}
          valueKey="value"
          valueRenderer={[Function]}
        />
        <DropDown
          className="caret"
          fill="currentColor"
          height={32}
          width={32}
        />
      </div>
    </div>
  </div>
</SingleSelect>
`;
